### PR TITLE
Remove generatorParameter and asyncParameter contexts.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Paging @rbuckton

File: src/compiler/types.ts [hunk 1]
```diff
@@ -376,18 +376,12 @@ namespace ts {
         // If this node was parsed in the 'yield' context created when parsing a generator.
         Yield = 1 << 2,
 
-        // If this node was parsed in the parameters of a generator.
-        GeneratorParameter = 1 << 3,
-
         // If this node was parsed as part of a decorator
         Decorator = 1 << 4,
```
Comment (rbuckton) [inline src/compiler/types.ts]: Should we also renumber these?

Comment (CyrusNajmabadi) [inline src/compiler/types.ts]: I was going to do that once the change to remove 'strict mode' gets merged into this.

File: tests/baselines/reference/FunctionDeclaration10_es6.errors.txt [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
 
 
-==== tests/cases/conformance/es6/functionDeclarations/FunctionDeclaration10_es6.ts (1 errors) ====
+==== tests/cases/conformance/es6/functionDeclarations/FunctionDeclaration10_es6.ts (5 errors) ====
     function * foo(a = yield => yield) {
-             ~
-!!! error TS1220: Generators are only available when targeting ECMAScript 6 or higher.
+                   ~~~~~~~~~
+!!! error TS2371: A parameter initializer is only allowed in a function or constructor implementation.
+                             ~~
```
Comment (rbuckton) [inline tests/baselines/reference/FunctionDeclaration10_es6.errors.txt]: It looks like the parser goes off the rails here.

File: tests/baselines/reference/asyncArrowFunction9_es6.errors.txt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+
+==== tests/cases/conformance/async/es6/asyncArrowFunction/asyncArrowFunction9_es6.ts (6 errors) ====
+    var foo = async (a = await => await): Promise<void> => {
+              ~~~~~
+!!! error TS2304: Cannot find name 'async'.
+                     ~
+!!! error TS2304: Cannot find name 'a'.
+                                        ~
+!!! error TS1005: ',' expected.
```
Comment (rbuckton) [inline tests/baselines/reference/asyncArrowFunction9_es6.errors.txt]: Same thing here, it looks like the parser goes off the rails.

**Pull request merged**